### PR TITLE
Sym upstream pr2 examples

### PR DESCRIPTION
**Pull request by ktangsali**
# PhysicsNeMo Pull Request

## Description
PR 2 of Sym upstream efforts. In this PR we port the examples to use the new Sym interface and drop the geometry specific imports from previous sym in support of PhysicsNeMo.Mesh + PyVista. 

Depends on https://github.com/NVIDIA/physicsnemo/pull/1567 (PhysicsInformer upstreaming - merged)

### Changes
* Import updates: Replace physicsnemo.sym.hydra, physicsnemo.sym.distributed, and physicsnemo.sym.eq.spatial_grads imports with their PhysicsNeMo equivalents
* Inline PDE definitions: Replace pre-built PDE classes (NavierStokes, Diffusion, IncompressibleNavierStokes) with inline SymPy definitions in each example
* Key/Arch removal: Replace Key objects and Arch base class with plain strings and torch.nn.Module
* Geometry replacement: Drop physicsnemo.sym.geometry imports in favor of physicsnemo.mesh primitives, sample_random_points_on_cells, and PyVista for STL loading; analytical SDF for axis-aligned box domains
* Derivative functional: Replace FirstDeriv with mesh_lsq_gradient from physicsnemo.nn.functional.derivatives in DoMINO physics loss
* Documentation: Update README, FAQ, example READMEs, and migration guide to reflect that PhysicsNeMo-Sym is archived and its functionality is now built-in

All changes have been verified by running the actual example (in case of LDC) or a minimal test to check the reproducibility of the behavior using the new code changes

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/cfd/external_aerodynamics/domino/src/loss.py [hunk 1]
```diff
@@ -32,20 +45,21 @@ def compute_physics_loss(
     mask: torch.Tensor,
     loss_type: Literal["mse", "rmse"],
     dims: tuple[int, ...] | None,
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/external_aerodynamics/domino/src/loss.py:47]: Since these are in two different directories, let's keep it as it.

File: examples/cfd/ldc_pinns/train.py [hunk 2]
```diff
@@ -73,92 +147,65 @@ def ldc_trainer(cfg: DictConfig) -> None:
...
-            optimizer.step()
-            scheduler.step()
+        optimizer.zero_grad()
+
+        bc_data = sample_boundary(2000, dist.device)
+        int_data = sample_interior(4000, dist.device)
+
+        y_vals = bc_data["y"]
+        mask_no_slip = y_vals < height / 2
+        mask_top_wall = y_vals >= height / 2 - 1e-7
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/ldc_pinns/train.py]: Done.

File: examples/cfd/darcy_physics_informed/darcy_physics_informed_fno.py [hunk 3]
```diff
@@ -23,14 +23,43 @@
...
+    Reference: https://en.wikipedia.org/wiki/Diffusion_equation
+    """
+
+    def __init__(self, T="T", D="D", Q=0, dim=2, time=False):
+        self.dim = dim
+        x, y = Symbol("x"), Symbol("y")
+        iv = {"x": x, "y": y}
+        T_var = Function(T)(*iv.values())
+        D_var = Function(D)(*iv.values()) if isinstance(D, str) else Number(D)
+        Q_var = Number(Q) if isinstance(Q, (int, float)) else Q
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/cfd/darcy_physics_informed/darcy_physics_informed_fno.py]: Done.

File: examples/cfd/datacenter/inference.py [hunk 4]
```diff
@@ -30,12 +30,58 @@
...
+import itertools
 import os
 import numpy as np
 from vtk.util.numpy_support import vtk_to_numpy, numpy_to_vtk
-from physicsnemo.sym.geometry.primitives_3d import Box, Channel
-from physicsnemo.sym.utils.io.vtk import var_to_polyvtk
-import itertools
+
+
+def _box_sdf(points, lower, upper):
```
Comment (loliverhennigh) [inline examples/cfd/datacenter/inference.py:39]: As a side note it would be really nice to have an SDF module in PhysicsNemo somehow. Like a very optimized one and put it in the `nn.functional` space. There are like tree search algorithms to make these kinds of SDF calculations really fast

Comment (coreyjadams) [reply examples/cfd/datacenter/inference.py:39]: https://github.com/NVIDIA/physicsnemo/blob/main/physicsnemo/nn/functional/geometry/sdf.py

??

**Review by loliverhennigh — APPROVED** [review]
LGTM

File: examples/cfd/datacenter/inference.py (hunk 4 above)
Comment (ktangsali) [reply examples/cfd/datacenter/inference.py:39]: Discussed offline, this would be a good to have, but not a blocker for this PR

File: examples/cfd/external_aerodynamics/domino/src/loss.py [hunk 5]
```diff
@@ -17,13 +17,26 @@
...
+    offsets_list = [0]
+    indices_list = []
+    for node_id in range(num_nodes):
+        if node_id in neighbors_list:
+            indices_list.extend(neighbors_list[node_id])
+        offsets_list.append(len(indices_list))
+    return (
+        torch.tensor(offsets_list, dtype=torch.int64, device=device),
+        torch.tensor(indices_list, dtype=torch.int64, device=device),
+    )
```
Comment (peterdsharpe) [inline examples/cfd/external_aerodynamics/domino/src/loss.py:39]: This op can be accelerated using `Mesh.get_point_to_points_adjacency()`, which returns an `Adjacency` object with `.offsets` and `.indices`. (First, construct a graph as a `Mesh[1, 3]`, and then call the Adjacency to do this.) 

To go further - the physics loss computation could actually entirely be done with `Mesh.calculus`, if you want.

Optional, but would reduce a lot of boilerplate throughout this file and add vectorization.

Comment (ktangsali) [reply examples/cfd/external_aerodynamics/domino/src/loss.py:39]: Thanks Peter, if I understand, the `Mesh.calculus` is already using the same functionals? Regarding the Adjacency, let me track that in a separate PR as I would like to test it first

**Comment by ktangsali** [comment]
Blossom CI passing. Merging

File: examples/cfd/external_aerodynamics/domino/src/loss.py (hunk 5 above)
Comment (peterdsharpe) [reply examples/cfd/external_aerodynamics/domino/src/loss.py:39]: Super fair, given code freeze
